stl/inc/functional

```diff
@@ -1479,6 +1479,83 @@ namespace placeholders {
     _INLINE_VAR constexpr _Ph<20> _20{};
 } // namespace placeholders
 
+#if _HAS_CXX20
+// FUNCTION TEMPLATE _Call_front_binder
+template <size_t... _Ix, class _Cv_FD, class _Cv_tuple_TiD, class... _Unbound>
+constexpr auto _Call_front_binder(index_sequence<_Ix...>, _Cv_FD&& _Obj, _Cv_tuple_TiD&& _Tpl,
+    _Unbound&&... _Unbargs) noexcept(noexcept(_STD invoke(_STD forward<_Cv_FD>(_Obj),
+    _STD get<_Ix>(_STD forward<_Cv_tuple_TiD>(_Tpl))..., _STD forward<_Unbound>(_Unbargs)...)))
+    -> decltype(_STD invoke(_STD forward<_Cv_FD>(_Obj), _STD get<_Ix>(_STD forward<_Cv_tuple_TiD>(_Tpl))...,
+        _STD forward<_Unbound>(_Unbargs)...)) {
+    return _STD invoke(_STD forward<_Cv_FD>(_Obj), _STD get<_Ix>(_STD forward<_Cv_tuple_TiD>(_Tpl))...,
+        _STD forward<_Unbound>(_Unbargs)...);
+}

```

Comment:
can we get some comments or formatting to make the boundaries between each component of this definition more clear?

Maybe split each parameter to the invoke in the noexcept(noexcept( on its own line?

Alliteratively maybe some using aliases could be helpful here. It took me a good 10min just to parse this definition.

Reply:
Empty comments `//` can be used to force clang-format to wrap lines, without disabling clang-format entirely.

I don't see how `using` could help. There is, unfortunately, no way to centralize these `forward` calls without macros. Also unfortunately, `decltype(auto)` doesn't SFINAE, and the Committee has steadfastly refused to add `noexcept(auto)`.

Reply:
Is it time to breakdown and define the `_THREEPEAT` macro?

```c++
#define _FWD(x) static_cast<decltype(x)>(x)
#define _THREEPEAT(x)     \
    noexcept(noexcept(x)) \
    -> decltype((x)) {    \
        return (x);       \
    }

template <size_t... _Ix, class _Cv_FD, class _Cv_tuple_TiD, class... _Unbound>
constexpr auto _Call_front_binder(index_sequence<_Ix...>, _Cv_FD&& _Obj, _Cv_tuple_TiD&& _Tpl,
    _Unbound&&... _Unbargs) 
    _THREEPEAT(_STD invoke(FWD(_Obj), _STD get<_Ix>(FWD(_Tpl))..., FWD(_Unbargs)...))
```

Reply:
While tempting, I believe that macroization interferes with the clarity of our code, in addition to potentially harming modularization.

Reply:
I used `_BINDER_OPERATOR` previously, so my thinking has clearly evolved.

Reply:
Is the consensus then to use the empty `//` comments to wrap the lines?
If so, how do we want them to be formatted?
e.g.
```
constexpr auto operator()(/* params */) //
/* refqualifier */ noexcept(noexcept()) //
-> decltype(/* */) {
    return //...
}
```

Reply:
Yeah, let’s try wrapping with comments. I think that’s a good skeleton, and if you put each argument on its own line (for decltype, noexcept, and return), the structure will be clearer, at the cost of vertical space which is OK.